File: 03-semester-c/01-coursework/01-variables-data-types-operators/01-basic-arithmetic.c
/*
 * Topic: Variables, Data Types, Operators & I/O
 * Sample 1: Basic Arithmetic Operations
 *
 * Demonstrates:
 * - Variable declaration and initialization
 * - Fundamental data types (int, float, double)
 * - Arithmetic operators (+, -, *, /, %)
 * - Input/output with printf() and scanf()
 * - Type casting and precision formatting
 */

#include <stdio.h>

int main()
{
    /* Variable declaration with appropriate data types */
    int num1, num2;
    float result;

    printf("=== Basic Arithmetic Calculator ===\n");

    /* Prompt user for first integer input */
    printf("Enter first number: ");
    if (scanf("%d", &num1) != 1) {
        printf("Error: Invalid input\n");
        return 1;
    }

    /* Prompt user for second integer input */
    printf("Enter second number: ");
    if (scanf("%d", &num2) != 1) {
        printf("Error: Invalid input\n");
        return 1;
    }

    /* Perform and display all arithmetic operations */
    printf("\nResults:\n");
    printf("%d + %d = %d\n", num1, num2, num1 + num2);
    printf("%d - %d = %d\n", num1, num2, num1 - num2);
    printf("%d * %d = %d\n", num1, num2, num1 * num2);

    /* Integer division truncates, so show both integer and float division */
    printf("%d / %d = %d\n", num1, num2, num1 / num2);
    printf("%d %% %d = %d\n", num1, num2, num1 % num2);

    /* Float division demonstrates type casting and precision */
    result = (float)num1 / num2;
    printf("Float division: %.2f / %.2f = %.2f\n", (float)num1, (float)num2, result);

  return 0;
}

File: 01-semester-python/01-coursework/01-python-basics-syntax/01-basic-arithmetic.py
"""
Topic: Python Basics and Syntax
Sample 1: Basic Arithmetic and Variable Operations

Demonstrates:
- Variable declaration and assignment
- Fundamental data types (int, float, str, bool)
- Arithmetic operators (+, -, *, /, //, %, **)
- Input/output with input() and print()
- String formatting and f-strings
"""


def main():
    print("=== Python Arithmetic Calculator ===\n")

    # Get user input (input() always returns a string)
    num1 = float(input("Enter first number: "))
    num2 = float(input("Enter second number: "))

    # Perform arithmetic operations
    print("\nResults:")
    print(f"{num1} + {num2} = {num1 + num2}")
    print(f"{num1} - {num2} = {num1 - num2}")
    print(f"{num1} * {num2} = {num1 * num2}")

    # Division with error handling
    if num2 != 0:
        print(f"{num1} / {num2} = {num1 / num2:.2f}")
        print(f"{num1} // {num2} = {int(num1 // num2)}")  # Floor division
        print(f"{num1} % {num2} = {num1 % num2:.2f}")  # Modulo
    else:
        print("Cannot divide by zero!")

    # Power operation
    print(f"{num1} ** {num2} = {num1 ** num2}")

    # Boolean expressions
    print(f"\nComparisons:")
    print(f"{num1} > {num2}: {num1 > num2}")
    print(f"{num1} == {num2}: {num1 == num2}")
    print(f"{num1} != {num2}: {num1 != num2}")


if __name__ == "__main__":
    main()
